packages/firestore/src/api/credentials.ts

```diff
@@ -306,6 +336,33 @@ export class FirebaseCredentialsProvider implements CredentialsProvider {
     );
     return new User(currentUid);
   }
+
+  /**
+   * Blocks the AsyncQueue until the next user is available. This is invoked
+   * on SDK start to wait for the first user token (or `null` if Auth is not yet
+   * loaded). If Auth is loaded after Firestore,
+   * `awaitTokenAndRaiseInitialEvent()` is also used to block Firestore until
+   * Auth is fully initialized.
+   *
+   * This function also invokes the change listener synchronously once a token
```

Comment:
Do you mean asynchronously, since it happens in the asyncQueue?

Reply:
I meant "synchronously with the code that resolves the token". I have reworded it.